lib/tasks/dicard_trashed_vacancies.rake

```diff
@@ -0,0 +1,6 @@
+namespace :vacancies do
+  desc "Discard trashed vacancies"
+  task discard_trashed: :environment do
+    Vacancy.trashed.or(Vacancy.removed_from_external_system).update_all(discarded_at: Time.current)
```

Comment:
If we try to be able to identify when a vacancy was trashed or removed from external system, could we use their `updated_at` instead of `Time.current` to set the `discarded_at` timestamp?

It may be imperfect, but I guess would be pretty accurate for the majority of vacancies.

Reply:
I see that the[ follow-up PR](https://github.com/DFE-Digital/teaching-vacancies/pull/7732/files) removes the `trashed` status (:raised_hands: ). 
Would it be appropriate to use this task to also backfill all the `trashed` to `published`? 
Or `draft` vacancies could also be trashed, and we can't distinguish if a current `trashed` should map to `published` or `draft`? :thinking:

Reply:
Thanks - I had intended to update the status to published, but forgot when copying the example from the discard docs. Don't think draft vacancies can be trashed - it's only external vacancies that can be soft deleted (in the current code base anyway)

Reply:
Great then. Thanks for looking into this!